src/App.tsx

```diff
@@ -1,14 +1,62 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+
+const baseTodo: Todo = {
+  id: 0,
+  title: '',
+  completed: false,
+  userId: 0,
+};
+
+const filterTodos = (sourceTodos: Todo[], filterBy: FilterType | string, currentQuery: string) => {
+  let filteredTodos: Todo[];
+
+  switch (filterBy) {
+    case FilterType.Active:
+      filteredTodos = sourceTodos.filter(todo => !todo.completed);
+      break;
+
+    case FilterType.Completed:
+      filteredTodos = sourceTodos.filter(todo => todo.completed);
+      break;
+
+    default:
+      filteredTodos = [...sourceTodos];
+      break;
+  }
+
+  if (currentQuery) {
+    filteredTodos = filteredTodos
+      .filter(todo => todo.title.toLowerCase().includes(currentQuery.toLowerCase()));
+  }
+
+  return filteredTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([baseTodo]);
+  const [selectedTodo, setSelectedTodo] = useState(baseTodo);
+  const [isListLoading, setIsListLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [filterByStatus, setFilterByStatus] = useState<string>(FilterType.All);
```

Comment:
```suggestion
  const [filterByStatus, setFilterByStatus] = useState<FilterType>(FilterType.All);
```
It's not a string, you have created enum for this filter